src/astgen

```diff
@@ -547,6 +553,57 @@ def parseOpType(string):
     return None
 
 
+def read_graph_types(fh, vertex_types, edge_types):
+    has_errors = False
+
+    def handle_match(match, types):
+        keywrd = match.group(1)
+        if keywrd == "struct":
+            print(
+                filename + ":" + str(lineno) +
+                ": %Error: struct not allowed in graph vertex/edge declaration"
+            )
+            has_errors = True
+        classn = match.group(2)
+        supern = match.group(3)
+        super = types.get(supern)
+        if not super:
+            super = Node(supern, None)
+            types[supern] = super
+        node = types.get(classn)
+        if node:
+            node.superClass = supern
+        else:
+            node = Node(classn, super)
+            types[classn] = node
+        if super:
+            super.addSubClass(node)
+
+    re_first_part = r'^\s*(class|struct)\s*([a-zA-Z0-9_]+'
+    re_second_part = r')\s+(?:final|VL_NOT_FINAL)\s*:\s*(?:public|protected|private)?\s+([a-zA-Z0-9_]+)'
+    special_vertices = '|'.join(
+        ["AbstractMTask", "AbstractLogicMTask", "ExecMTask", "LogicMTask"])
```

Comment:
What determines "special" (document it).  Can we instead put some magic annotation in the C++ source code this detects?